resources-compose/src/androidLibMain/kotlin/GradientStyle.kt

```diff
@@ -0,0 +1,211 @@
+        val centerY = center.y * size.height
+
+        return RadialGradientShader(
+            colors = colors,
+            colorStops = stops,
+            center = Offset(centerX, centerY),
+            radius = if (radius == Float.POSITIVE_INFINITY) size.minDimension / 2 else radius,
+            tileMode = tileMode
+        )
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is RelativeRadialGradient) return false
+
+        if (colors != other.colors) return false
+        if (stops != other.stops) return false
+        if (center != other.center) return false
+        if (radius != other.radius) return false
+        if (tileMode != other.tileMode) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = colors.hashCode()
+        result = 31 * result + (stops?.hashCode() ?: 0)
+        result = 31 * result + center.hashCode()
+        result = 31 * result + radius.hashCode()
+        result = 31 * result + tileMode.hashCode()
+        return result
+    }
+
+    override fun toString(): String {
+        val radiusValue = if (radius.isFinite()) "radius=$radius, " else ""
+        return "RelativeRadialGradient(" +
+            "colors=$colors, " +
+            "stops=$stops, " +
+            "center=$center, " +
+            radiusValue +
+            "tileMode=$tileMode)"
+    }
+}
+
+/**
+ * Brush implementation used to apply a sweep gradient on a given [Paint]
+ */
+@Immutable
+class RelativeSweepGradient internal constructor(
+    private val center: GradientStyle.CenterPoint,
+    private val colors: List<Color>,
+    private val stops: List<Float>? = null
+) : ShaderBrush() {
+
+    constructor(center: GradientStyle.CenterPoint, vararg colorStops: Pair<Float, Color>) : this(
+        colors = List<Color>(colorStops.size) { i -> colorStops[i].second },
+        stops = List<Float>(colorStops.size) { i -> colorStops[i].first },
+        center = center
+    )
+
+    override fun createShader(size: Size): Shader =
+        SweepGradientShader(
+            Offset(
+                center.x * size.width,
+                center.y * size.height
+            ),
+            colors,
+            stops
+        )
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is RelativeSweepGradient) return false
+
+        if (center != other.center) return false
+        if (colors != other.colors) return false
+        if (stops != other.stops) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = center.hashCode()
+        result = 31 * result + colors.hashCode()
+        result = 31 * result + (stops?.hashCode() ?: 0)
+        return result
+    }
+
+    override fun toString(): String {
+        return "RelativeSweepGradient(" +
+            "center=$center, " +
+            "colors=$colors, stops=$stops)"
+    }
+}
```

Comment:
Would be nice to have some @preview s in this file, but can be also covered by the ones in BackgroundStyle

Reply:
We can add this as a incremental improvement, resolving for now